### PR TITLE
Add support for filtering by directory

### PR DESCRIPTION
**Pull request by jaredly**
## Summary:
webapp does some checking by directory to know whether to rerun jest on all files. This adds support for that.

Issue: XXX-XXXX

## Test plan:
I've added a test! 🤞

### CODE REVIEW

File: .github/workflows/node-ci.yml [hunk 1]
```diff
@@ -12,3 +12,38 @@ jobs:
       - uses: ./actions/check-for-changeset
       - uses: ./actions/shared-node-cache
       - run: yarn lint
+
+      # Test filter-files
+      - uses: ./actions/filter-files
+        id: fftest1
+        with:
+          changed-files: '["one", "two/three"]'
+          files: 'one, two, three'
```
Comment (kevinbarabash) [inline .github/workflows/node-ci.yml:21]: I know this is covered in the docs, but could you add a test case with the same `changed-files`, but with `files: 'two/three'` to illustrate that it matches exact file paths?

**Review by kevinbarabash — APPROVED** [review]
LGTM.  Clever way of testing the action using a workflow.

**Review by somewhatabstract — APPROVED** [review]
Thanks for adding this. very useful!

File: actions/filter-files/action.yml [hunk 2]
```diff
@@ -2,10 +2,10 @@ name: 'Filter files please'
 description: 'Filter the list of changed files'
 inputs:
   changed-files:
-    description: 'jsonified list of changed files from setup'
+    description: 'jsonified list of changed files'
     required: true
   files:
-    description: 'comma-separated list of files to check for'
+    description: 'comma-separated list of files to match against. names ending in "/" will be treated as a directory, and match as a prefix. Otherwise path names must be a full match.'
```
[the remark this answers is not in the document]
Comment (jaredly) [reply actions/filter-files/action.yml:8]: yes that's right

**Pull request merged**